frontend/src/components/Forms/reports/ReportFilterForm.js

```diff
@@ -0,0 +1,216 @@
+  selectedMineReportCategory: PropTypes.string.isRequired,
+  selectedMineReportDefinitionGuid: PropTypes.string.isRequired,
+};
+
+const defaultProps = {};
+
+const selector = formValueSelector(FORM.FILTER_REPORTS);
+
+export class ReportFilterForm extends Component {
+  state = {
+    dropdownMineReportDefinitionOptionsFiltered: [],
+    dropdownMineReportCategoryOptionsFiltered: [],
+  };
+
+  handleReset = () => {
+    this.props.reset();
+    this.props.onSubmit();
+  };
+
+  componentDidMount = () => {
+    this.updateMineReportOptions(this.props.mineReportDefinitionOptions);
+    this.updateMineReportCategoryOptions(this.props.dropdownMineReportCategoryOptions);
+  };
+
+  updateMineReportOptions = (mineReportDefinitionOptions, selectedMineReportCategory) => {
+    let mineReportDefinitionOptionsFiltered = mineReportDefinitionOptions;
+
+    if (selectedMineReportCategory) {
+      mineReportDefinitionOptionsFiltered = mineReportDefinitionOptions.filter(
+        (rd) =>
+          rd.categories.filter((c) => c.mine_report_category === selectedMineReportCategory)
+            .length > 0
+      );
+    }
+
+    const dropdownMineReportDefinitionOptionsFiltered = createDropDownList(
+      mineReportDefinitionOptionsFiltered,
+      "report_name",
+      "mine_report_definition_guid"
+    );
+
+    this.setState({
+      dropdownMineReportDefinitionOptionsFiltered,
+    });
+  };
+
+  updateMineReportCategoryOptions = (
+    dropdownMineReportCategoryOptions,
+    selectedMineReportDefinitionGuid
+  ) => {
+    let dropdownMineReportCategoryOptionsFiltered = dropdownMineReportCategoryOptions;
+
+    if (selectedMineReportDefinitionGuid) {
+      const selectedMineReportDefinition = this.props.mineReportDefinitionOptions.filter(
+        (option) => option.mine_report_definition_guid === selectedMineReportDefinitionGuid
+      )[0];
+
+      dropdownMineReportCategoryOptionsFiltered = dropdownMineReportCategoryOptions.filter((cat) =>
+        selectedMineReportDefinition.categories
+          .map((category) => category.mine_report_category)
+          .includes(cat.value)
+      );
+    }
+    this.setState({
+      dropdownMineReportCategoryOptionsFiltered,
+    });
+  };
+
+  componentWillReceiveProps = (nextProps) => {
+    if (nextProps.selectedMineReportCategory !== this.props.selectedMineReportCategory) {
+      this.updateMineReportOptions(
+        nextProps.mineReportDefinitionOptions,
+        nextProps.selectedMineReportCategory
+      );
+    }
+    if (
+      nextProps.selectedMineReportDefinitionGuid !== this.props.selectedMineReportDefinitionGuid
+    ) {
+      this.updateMineReportCategoryOptions(
+        nextProps.dropdownMineReportCategoryOptions,
+        nextProps.selectedMineReportDefinitionGuid
+      );
+    }
+  };
+
+  render() {
+    return (
+      <Form
+        layout="vertical"
+        onSubmit={this.props.handleSubmit}
+        onReset={this.handleReset}
+        enableReinitialize="true"
+      >
```

Comment:
`enableReinitialize` has already been passed into the form, this line shouldn't be needed
 ```reduxForm({
    form: FORM.FILTER_REPORTS,
    touchOnBlur: true,
    enableReinitialize: true,
  })
```

Reply:
I was trying to figure out where to add it and forgot to remove this. Thanks.